common/src/services/userVerification.service.ts

```diff
@@ -15,16 +12,19 @@ import { SecretVerificationRequest } from '../models/request/secretVerificationR
 
 import { Verification } from '../types/verification';
 
-@Injectable()
 export class UserVerificationService implements UserVerificationServiceAbstraction {
     constructor(private cryptoService: CryptoService, private i18nService: I18nService,
-        private platformUtilsService: PlatformUtilsService, private apiService: ApiService,
-        private logService: LogService) { }
+        private platformUtilsService: PlatformUtilsService, private apiService: ApiService) { }
 
     async buildRequest<T extends SecretVerificationRequest>(verification: Verification,
         requestClass?: new () => T, alreadyHashed?: boolean) {
         if (verification?.secret == null || verification.secret === '') {
-            throw new Error('No secret provided for verification.');
+            if (verification.type === VerificationType.OTP) {
+                this.handleError(this.i18nService.t('verificationCodeRequired'));
+            } else {
+                this.handleError(this.i18nService.t('masterPassRequired'));
+            }
+            return null;
```

Comment:
I'm slightly hesitant to this change, previously we threw an error which got handled by the components, do we plan on updating each component to do a null check?

It feels a bit like a side effect to have the `buildRequest` create a toast.

Reply:
This is done